src/xrc/src/forex.rs

```diff
@@ -288,48 +289,63 @@ impl ForexRateStore {
+        while go_back_days <= MAX_DAYS_TO_GO_BACK {
+            let query_timestamp = timestamp - SECONDS_PER_DAY * go_back_days;
+            go_back_days += 1;
+            if let Some(rates_for_timestamp) = self.rates.get(&query_timestamp) {
+                let base = rates_for_timestamp.get(&base_asset);
+                let quote = rates_for_timestamp.get(&quote_asset);
+
+                match (base, quote) {
+                    (Some(base_rate), Some(quote_rate)) => {
+                        return Ok(base_rate.clone() / quote_rate.clone())
                     }
-                }
-                (None, Some(_)) => Err(GetForexRateError::CouldNotFindBaseAsset(
-                    timestamp,
-                    base_asset.to_string(),
-                )),
-                (None, None) => {
-                    if quote_asset == USD {
-                        Err(GetForexRateError::CouldNotFindBaseAsset(
-                            timestamp,
-                            base_asset.to_string(),
-                        ))
-                    } else {
-                        Err(GetForexRateError::CouldNotFindAssets(
+                    (Some(base_rate), None) => {
+                        // If the quote asset is USD, it should not be present in the map and the base rate already uses USD as the quote asset.
+                        if quote_asset == USD {
+                            return Ok(base_rate.clone());
+                        } else {
+                            return Err(GetForexRateError::CouldNotFindQuoteAsset(
+                                timestamp,
+                                quote_asset.to_string(),
+                            ));
+                        }
+                    }
+                    (None, Some(_)) => {
+                        return Err(GetForexRateError::CouldNotFindBaseAsset(
                             timestamp,
                             base_asset.to_string(),
-                            quote_asset.to_string(),
                         ))
                     }
+                    (None, None) => {
+                        if quote_asset == USD {
+                            return Err(GetForexRateError::CouldNotFindBaseAsset(
+                                timestamp,
+                                base_asset.to_string(),
+                            ));
+                        } else {
+                            return Err(GetForexRateError::CouldNotFindAssets(
+                                timestamp,
+                                base_asset.to_string(),
+                                quote_asset.to_string(),
+                            ));
+                        }
+                    }
                 }
+            } else {
+                // No forex rates were found for the query timestamp.
+                continue;
```

Comment:
Is the continue necessary? It seems pretty clear that no rates were found and the loop is going to continue.

Reply:
Yes you are right, it is not necessary at all.